onnxruntime/core/optimizer/graph_optimizer_registry.cc

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/optimizer/graph_optimizer_registry.h"
+#include "core/optimizer/graph_transformer_utils.h"
+#include "core/optimizer/selection_and_optimization_func.h"
+#include "core/optimizer/qdq_transformer/constant_folding_dq_node.h"
+
+using namespace onnxruntime;
+using namespace ::onnxruntime::common;
+
+namespace onnxruntime {
+GraphOptimizerRegistry::GraphOptimizerRegistry(const onnxruntime::SessionOptions* sess_options,
+                                               const onnxruntime::IExecutionProvider* cpu_ep,
+                                               const logging::Logger* logger) : session_options_(sess_options),
+                                                                                cpu_ep_(cpu_ep),
+                                                                                logger_(logger) {
+  auto status = CreatePredefinedSelectionFuncs();
+  ORT_ENFORCE(status.IsOK(), "Could not create pre-defined selection functions. Error Message: ",
+              status.ErrorMessage());
+}
+
+Status GraphOptimizerRegistry::CreatePredefinedSelectionFuncs() {
+#if !defined(ORT_MINIMAL_BUILD)
```

Comment:
Should we make the whole class a stub for a minimal build so there's no binary size cost? 
Probably want to enable it for an extended minimal build though as that allows extra EPs to be enabled.

i.e. use `#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)` to exclude all the implementation so there's a type defined but it's empty. May need a dummy var like this to make all the compilers happy.

https://github.com/microsoft/onnxruntime/blob/9d0dc9f062fa496422c3679204ecb62c65430758/include/onnxruntime/core/session/onnxruntime_c_api.h#L4998-L5002

Reply:
Hmm, the graph partitioner is created in `PartitionOrtFormatModel` where it needs the GraphOptimizerRegistry instance and this part is in minimal build. So, i think it's hard to make whole class a stub for a minimal build.

Reply:
Discussed offline. Made the GraphOptimizerRegistry class a stub for minimal build.

For the extended minimal build, do we want to support ConstantFoldingDQ optimizer?
ConstantFoldingDQ relies on ConstantFolding which has a bunch of graph/node utils, e.g. `InlineIfSubgraph`, `RemoveNodesWithOneOutputBottomUp`, `IsOperationDeterministic` ... that needs to be moved to ifdef as well, so i dont' think ConstantFoldingDQ is suitable for extended minimal build.